src/otx/core/data/dataset/tile.py

```diff
@@ -152,20 +229,30 @@ class OTXTileTrainDataset(OTXTileDataset):
         tile_config (TilerConfig): Tile configuration.
     """
 
-    def __init__(self, dataset: OTXDataset, tile_config: TilerConfig) -> None:
+    def __init__(self, dataset: OTXDataset, tile_config: TileConfig) -> None:
         dm_dataset = dataset.dm_subset.as_dataset()
         dm_dataset = dm_dataset.transform(
-            Tile,
-            grid_size=tile_config.grid_size,
+            OTXTileTransform,
+            tile_size=tile_config.tile_size,
             overlap=(tile_config.overlap, tile_config.overlap),
             threshold_drop_ann=0.5,
         )
         dm_dataset = dm_dataset.filter("/item/annotation", filter_annotations=True, remove_empty=True)
-        dm_subset = DatasetSubset(dm_dataset, dataset.dm_subset.name)
-        dataset.dm_subset = dm_subset
-        dataset.ids = [item.id for item in dm_subset]
+        self.dm_subset = DatasetSubset(dm_dataset, dataset.dm_subset.name)
+        self.tile_config = tile_config
+        sampled_subset = self.sample_subset()
+        dataset.dm_subset = sampled_subset
+        dataset.ids = [item.id for item in sampled_subset]
         super().__init__(dataset, tile_config)
 
+    def sample_subset(self) -> DatasetSubset:
```

Comment:
Why do we need this?

Reply:
It's used in `TileSamplingHook` which sample tiles from training dataset when epoch starts